pkg/controller/common/manifestprocessing.go

```diff
@@ -78,22 +79,22 @@ func (p *ManifestProcessor) ProcessManifests(manifests []manifest.Manifest, comp
 	return utilerrors.NewAggregate(allErrors)
 }
 
-func (p *ManifestProcessor) processObject(obj *unstructured.Unstructured, component string) error {
-	origLogger := p.Log
-	defer func() { p.Log = origLogger }()
+func (p *ManifestProcessor) processObject(ctx context.Context, obj *unstructured.Unstructured, component string) error {
+	origLogger := LogFromContext(ctx)
 
 	key := v1.NewResourceKey(obj, obj)
-	p.Log = origLogger.WithValues("Resource", key)
+	log := origLogger.WithValues("Resource", key)
+	ctx = NewContextWithLog(ctx, log)
```

Comment:
Can we simplify this to `ctx = NewContextWithLog(ctx, LogFromContext(ctx).WithValues("Resource", key)`?

Reply:
It wasn't possible directly, but I've now refactored the code.